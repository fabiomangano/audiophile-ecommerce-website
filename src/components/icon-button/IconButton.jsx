import React from 'react';
import PropTypes from 'prop-types';
import style from './IconButton.module.scss';

const IconButton = ({ children, onClick }) => {
  return (
    <button type="button" onClick={onClick} className={style.icon_btn}>
      {children}
    </button>
  );
};

IconButton.defaultProps = {
  children: null,
  onClick: undefined,
};

IconButton.propTypes = {
  children: PropTypes.node,
  onClick: PropTypes.func,
};

export default IconButton;
