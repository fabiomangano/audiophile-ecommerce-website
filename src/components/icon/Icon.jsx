import React from 'react';
import PropTypes from 'prop-types';
import icons from './icons';

export const DIRECTIONS = {
  TOP: 0,
  RIGHT: 90,
  BOTTOM: 180,
  LEFT: -90,
};

export const ICONS = icons;

const Icon = ({ path, color, size, direction }) => {
  return (
    <svg width={size} height={size} viewBox="0 0 24 24">
      <g transform={`rotate(${direction}, 12, 12)`}>
        <path fill={color} d={path} />
      </g>
    </svg>
  );
};

Icon.propTypes = {
  path: PropTypes.string.isRequired,
  color: PropTypes.string,
  size: PropTypes.number,
  direction: PropTypes.number,
};

Icon.defaultProps = {
  size: 24,
  color: 'currentColor',
  direction: DIRECTIONS.TOP,
};

export default Icon;
