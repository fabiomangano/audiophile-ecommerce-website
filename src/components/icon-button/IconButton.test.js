import React from 'react';
import { mount } from 'enzyme';
import ShallowRenderer from 'react-test-renderer/shallow';
import IconButton from './IconButton';
import Icon from '../icon/Icon';

jest.mock('../icon/Icon', () => () => <svg />);

describe('IconButton', () => {
  it('should renders correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<IconButton />);
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });

  it('should render the given icon', () => {
    const wrapper = mount(
      <IconButton>
        <Icon />
      </IconButton>,
    );

    expect(wrapper.find('svg').length).toEqual(1);
  });

  it('should call onClick function', () => {
    const onClick = jest.fn();
    const wrapper = mount(
      <IconButton onClick={onClick}>
        <Icon />
      </IconButton>,
    );

    wrapper.find('button').simulate('click');
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
