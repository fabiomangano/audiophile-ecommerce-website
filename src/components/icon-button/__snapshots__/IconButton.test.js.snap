// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton should renders correctly 1`] = `
<button
  className="icon_btn"
  type="button"
/>
`;
